modules/images/webp-uploads/load.php

```diff
@@ -5,40 +5,294 @@
+		return new WP_Error( 'image_file_not_present', __( 'The file key is not present on the image data', 'performance-lab' ) );
 	}
 
-	// Skip conversion when creating the `-scaled` image (for large image uploads).
-	if ( preg_match( '/-scaled\..{3}.?$/', $filename ) ) {
-		return $output_format;
+	if ( ! isset( $metadata['sizes'][ $size ]['sources'] ) || ! is_array( $metadata['sizes'][ $size ]['sources'] ) ) {
+		$metadata['sizes'][ $size ]['sources'] = array();
 	}
 
-	$output_format['image/jpeg'] = 'image/webp';
+	$metadata['sizes'][ $size ]['sources'][ $mime ] = array(
+		'file'     => $image['file'],
+		'filesize' => array_key_exists( 'path', $image ) ? filesize( $image['path'] ) : 0,
+	);
+
+	return wp_update_attachment_metadata( $attachment_id, $metadata );
+}
+
+add_action( 'webp_uploads_create_image', 'webp_uploads_generate_image_size', 10, 3 );
+
+/**
+ * Returns an array with the list of valid mime types that a specific mime type can be converted into it,
+ * for example an image/jpeg can be converted into an image/webp.
+ *
+ * @since n.e.x.t
+ *
+ * @return array<string, array<string>> An array of valid mime types, where the key is the mime type and the value is the extension type.
+ */
+function webp_uploads_get_supported_image_mime_transforms() {
+	$image_mime_transforms = array(
+		'image/jpeg' => array( 'image/webp' ),
+		'image/webp' => array( 'image/jpeg' ),
+	);
 
-	return $output_format;
+	/**
+	 * Filter to allow the definition of a custom mime types, in which a defined mime type
+	 * can be transformed and provide a wide range of mime types.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @param array<string, array<string>> An array with the valid mime transforms.
+	 *
+	 * @return array<string, array<string>> An array with the valid mime transformation
+	 */
+	return (array) apply_filters( 'webp_uploads_supported_image_mime_transforms', $image_mime_transforms );
+}
+
+/**
+ * Hook fired when an attachment is deleted, this hook is in charge of removing any
+ * additional mime types created by this plugin besides the original image. Any source
+ * with the same as the main image would not be removed by this hook due this file would
+ * be removed by WordPress when the attachment is deleted, usually this happens after this
+ * hook is executed.
+ *
+ * @since n.e.x.t
+ *
+ * @see   wp_delete_attachment
+ *
+ * @param int $attachment_id The ID of the attachment the sources are going to be deleted.
+ *
+ * @return void
+ */
+function webp_uploads_remove_sources_files( $attachment_id ) {
+	$metadata = wp_get_attachment_metadata( $attachment_id );
+
+	if (
+		! isset( $metadata['sizes'], $metadata['file'] )
+		|| empty( $metadata['file'] )
+		|| ! is_array( $metadata['sizes'] )
+	) {
+		return;
+	}
+
+	$upload_path = wp_get_upload_dir();
+	if ( empty( $upload_path['basedir'] ) ) {
+		return;
+	}
+
+	$file_directory = dirname( $metadata['file'] );
+	$directory      = path_join( $upload_path['basedir'], $file_directory );
```

Comment:
Nit-pick, but IMO it would be a good idea to stick to WP core names (see `wp_delete_attachment_files()` function) to clarify the variable names:
```suggestion
	$intermediate_dir = path_join( $upload_path['basedir'], dirname( $metadata['file'] ) );
```